src/ProductConstructionService/ProductConstructionService.WorkItems/WorkItemScope.cs

```diff
@@ -19,6 +19,7 @@ public class WorkItemScope : IAsyncDisposable
     private readonly Func<Task> _finalizer;
     private readonly IServiceScope _serviceScope;
     private readonly ITelemetryRecorder _telemetryRecorder;
+    private readonly List<IDisposable> _scopeDisposables;
```

Comment:
This looks sketchy. At the least, it should be a `Stack<T>` so that we dispose in the right order. But I am trying to understand why we had to do this?

Reply:
Last week we talked about how we wanted to extend the TelemetryScope to capture error logs too. Before, we'd dispose the scope after the `RunWorkItemAsync` method, which would happen before error handling. Now we dispose of the scope later

Reply:
But can't we just create the telemetry scope in `WorkItemConsumer`? Same goes for the telemetry operation (`StartOperation`).

Then we can maybe avoid these shenanigans?

Reply:
yeah we could, I thought this was more elegant, but I can change it if you think it's finicky